public/app/features/datasources/state/actions.ts

```diff
@@ -178,7 +178,10 @@ export const testDataSource = (
 export function loadDataSources(): ThunkResult<Promise<void>> {
   return async (dispatch) => {
     dispatch(dataSourcesLoad());
-    const response = await api.getDataSources();
+    let response: DataSourceSettings[] = [];
+    if (contextSrv.hasPermission(AccessControlAction.DataSourcesRead)) {
+      response = await api.getDataSources();
+    }
```

Comment:
What will happen now it that loading state will go from true to false, even though data sources were never loaded because of missing permissions. Is that what you want? Maybe do an early return instead? 

```typecript 
if (!contextSrv.hasPermission(AccessControlAction.DataSourcesRead)) {
  return;
}
```